### DISCUSSION TITLE
Adding Gutenberg Support to Pods Templates

### DISCUSSION THREAD

**Discussion by eristavi** [pods-framework/pods#6797] [Ideas]
## Problem
Currently, Pods Template System has a nice html editor what I find very intuitive however, now that we have an option for using Gutenberg blocks, I believe Pods Templates could benefit from it very much.

Are there any specific thoughts about it for future release?

**Comment by JoryHogeveen** [comment]
Hi @eristavi 

Thanks for your feature request.
While I can imagine the benefit of this, it would be difficult integrating the block editor with our custom magic tags etc.
Various blocks wouldn't be compatible with our magic tags (dynamic tags).

Still, I do like the idea so I will add it in the pipeline for future discussion (after Pods 2.8 at least).

**Comment by sc0ttkclark** [comment]
Definitely something to think about in the future. With the new Pods Blocks, it's kind of the best of both worlds there too.

We'll have to explore this idea further a bit later.

**Comment by sc0ttkclark** [comment]
I have some initial code for a Block Editor powered Pods Template screen. It's going to take more work to get it finished and make sure it can support new Pods conditional/each blocks that match the shortcodes.

**Comment by ehrenstein-c** [comment]
Given that the Gutenberg editor is the standard for WordPress content creation, there really is no alternative to providing a Gutenberg-compatible interface in the long run IMHO.
People quite naturally gravitate towards the solutions that will allow them to get away with the smallest possible toolset (and having to learn the fewest possible means to operate it).
I've tried to pitch Pods to intermediate WP course students, and they're already waving it off because of the lack of Gutenberg support.
Furthermore, the intuitiveness and beauty of working with Gutenberg blocks matches the beautiful concept of Pods quite well. ;-)

I see the pickle you're in, since it will be very hard, if not impossible, to migrate Pods to the "Gutenberg way" in a backwards-compatible manner.

**Comment by JoryHogeveen** [comment]
@Ehrenstein 

> Given that the Gutenberg editor is the standard for WordPress content creation, there really is no alternative to providing a Gutenberg-compatible interface in the long run IMHO.

Gutenberg is still only partially supported by many tools, including even WooCommerce which is maintained by Automattic.
While I fully understand (and support) your thoughts, Gutenberg/Blockeditor itself isn't there yet..

> I've tried to pitch Pods to intermediate WP course students, and they're already waving it off because of the lack of Gutenberg support.

What do you mean here? We fully support Gutenberg. Only the Pods Templates are code based. We even include blocks that you can use.

In general. Our templates are simply post types so enabling Gutenberg isn't the issue here. The issue is the amount of blocks available and the limitations in dynamic functionality of these blocks. Most blocks are simply static content with some UI editing capability.
Enabling Gutenberg should be done only if we are able to fully control how and what blocks are used. Otherwise it will only raise more issues than actually help people.

**Reply by ehrenstein-c** [reply]
Thanks for the clarification!
My comment about a "lack of Gutenberg support" lacked precision – I meant the lack of a means to setup Pods that fits well into the Gutenberg blocks ecosystem.

Fully controlling "how and what blocks are used" seems a bit of a stretch… but maybe I got the wrong idea.
In my experience using Pods requires a good deal of thought as well as trial-and-error, so I wouldn't mind some kind of "use with caution" approach to a full Gutenberg implementation. I don't consider Gutenberg blocks as the silver bullet, either (and I agree with your observation that the Blockeditor "itself isn't there yet.").

**Reply by sc0ttkclark** [reply]
Pods will be diving into more blocks oriented features soon but there are awesome features in many plugins out there which specialize in blocks themselves.

Pods specializes in content types and custom fields. You can create blocks with a few lines of code with out Pods Blocks API as well since Pods 2.8: https://docs.pods.io/code/blocks-api/

There are many new things being worked on including a Pods Template editor powered by blocks themselves which can be more easily used with PHP and other areas that templates can be consumed.
